### ISSUE TITLE
compose does not use default pasta netowrk 

### ISSUE THREAD

**Issue by az-z** [containers/podman-compose#967]
**Describe the bug**
i can't figure out how to make compose to use the pasta network mode.

please try to reproduce the bug in latest devel branch

**To Reproduce**
podman-compose version 1.1.0
podman version 4.9.4

Fedora release 39 (Thirty Nine)

I reset the system and specified the default network mode ( is it mode ? or is it a driver? ) to pasta:

```
[sailtech@dell5000 ~]$  grep cmd ~/.config/containers/containers.conf 
#default_rootless_network_cmd = "slirp4netns"
default_rootless_network_cmd = "pasta"
#network_cmd_path = ""
#network_cmd_options = []

[sailtech@dell5000 ~]$  podman info | grep networkBackend
  networkBackend: netavark
  networkBackendInfo:

[sailtech@dell5000 ~]$ podman run  -d --name=myubi registry.access.redhat.com/ubi8/ubi
[sailtech@dell5000 ~]$ podman inspect --format {{.HostConfig.NetworkMode}} myubi
pasta

[sailtech@dell5000 ~]$ podman-compose --in-pod=yes -f ./podman-compose.yml up
....
[sailtech@dell5000 ~]$ podman inspect --format {{.HostConfig.NetworkMode}} sailtech_restapi_1
bridge

[sailtech@dell5000 ~]$ podman network  ls
NETWORK ID    NAME              DRIVER
2f259bab93aa  podman            bridge
8f1d4e09447a  sailtech_default  bridge

[sailtech@dell5000 ~]$ podman network inspect sailtech_default
[
     {
          "name": "sailtech_default",
          "id": "8f1d4e09447a973e4225ba5262e4405722c480c42dcf70da2934fae2e9c1a91b",
          "driver": "bridge",
          "network_interface": "podman2",
          "created": "2024-06-18T23:07:06.718877522-04:00",
          "subnets": [
               {
                    "subnet": "10.89.1.0/24",
                    "gateway": "10.89.1.1"
               }
          ],
          "ipv6_enabled": false,
          "internal": false,
          "dns_enabled": true,
          "labels": {
               "com.docker.compose.project": "sailtech",
               "io.podman.compose.project": "sailtech"
          },
          "ipam_options": {
               "driver": "host-local"
          }
     }
]
```
it looks like the compose creates a bridge network ignoring the default setting to pasta.

the addition to compose file :
```
networks:
   default:
     mode: pasta
```
has no bearing on the execution result.

**Expected behavior**
i expect the compose to use the (default) pasta network 

**Actual behavior**
it doesn't

**Comment by jaimehrubiks** [comment]
Same issue here, I'm having issues creating rootless containers because (i think) of this error.

```
netavark: code: 3, msg: modprobe: ERROR: could not insert 'ip_tables': Operation not permitted
iptables v1.8.10 (legacy): can't initialize iptables table `nat': Table does not exist (do you need to insmod?)
Perhaps iptables or your kernel needs to be upgraded.
```

**Comment by jaimehrubiks** [comment]
I also found that this issue happens with podman as well, without compose

This works:
```
podman run -d -p 8000:8000/udp ubuntu:latest sleep infinity
```

This doesn't work:
```
podman network create infra
podman run --network=infra -d -p 8000:8000/udp ubuntu:latest sleep infinity
```
```
Error: netavark: iptables: No such file or directory (os error 2)
```

**Comment by az-z** [comment]
Missing package perhaps?

**Comment by jaimehrubiks** [comment]
Well, you are right in that, but it's still confusing.

So, if I configure the system as this using root:
```
dnf install netavark # this would also install dependencies: iptables-libs, iptables-legacy and iptables-legacy-libs
modprobe ip_tables
```
Then, it works, but I'm not sure if it's using Pasta

```
podman inspect 28c2d0e259f2acc96287552db59d7bd788b140fcfb818fe9b17a81cf30c2a9c4 | grep Net
          "NetworkSettings": {
               "Networks": {
                         "NetworkID": "infra",
               "NetworkMode": "bridge",
```

**Comment by az-z** [comment]
I could never see "pasta". But all my configurations are working. I have
inter comms and inbound connections. I have not tried creating new networks
or internetwork connections.

**Comment by jaimehrubiks** [comment]
I think that your description is accurate. It's basically working (if you have the right packages, which I was missing in my first experiments -iptables and ip_tables kernel module-), **but it doesn't use pasta**.

**Comment by az-z** [comment]
Compose is all another beast. Brace yourself.

**Comment by kavishgr** [comment]
> I also found that this issue happens with podman as well, without compose
> 
> This works:
> 
> ```
> podman run -d -p 8000:8000/udp ubuntu:latest sleep infinity
> ```
> 
> This doesn't work:
> 
> ```
> podman network create infra
> podman run --network=infra -d -p 8000:8000/udp ubuntu:latest sleep infinity
> ```

Same here. A `podman run` command uses **Pasta**, but it doesn't with a custom network. I don't think it's an issue related to Compose. I'm using `docker-compose` itself (with the Podman socket enabled and the DOCKER_HOST environment variable set). **I suspect it relates to a custom network**, as Compose also creates a custom network.

**Comment by qhaas** [comment]
Getting pasta to work with podman-compose is now more important since I've started having dns issues that are present in podman-compose launched containers, but not the same container launched with podman, with a recent update to Rocky 9.

**Comment by kavishgr** [comment]
Little update: I migrated to quadlets gradually. This wasn't because it's difficult, but more like taking on a project at a time, and it's the best thing I ever did. For those still on Compose, do give quadlet a try.

**Comment by eriksjolund** [comment]
Support for Slirp4netns will be removed in Podman 6 according to 

 [*Podman 6 High Level Design*](https://github.com/containers/podman/blob/main/contrib/design-docs/podman6hld.md)

**Comment by bam80** [comment]
Any progress? I think my problem is relative: #1429 
Can't run the container, had to disable network at all.

**Comment by daniyl-x** [comment]
_podman-compose version 1.5.0
podman version 5.4.2_

I have the same issue, it seems that podman-compose defaults to a bridge network mode, which results in both DNS and Internet connections not working for most of my rootless containers.

I was able to workaround that by using `network_mode: pasta` in my compose file, but doesn't cover the cases where you need to define a network in the compose file.

This is definitely a significant issue for my use case

**Comment by daniyl-x** [comment]
After some testing I discovered that the issue is not with podman-compose or pasta, which is confirmed by this response in the related Podman discussion: <https://github.com/containers/podman/discussions/24290#discussioncomment-10957302>.
As far as I understand, Podman doesn't allow (and probably isn't supposed to) using Podman-specific network backends (pasta, slirp4net) when creating networks with `podman network create mynet`.

I wasn't able to find the issue, but resetting podman with `podman system reset` (back up your volumes, containers, etc.) resolved it for me.

I believe that was likely a bug, probably a package management issue, caused by upgrade between major podman versions on my system.

**Comment by laur89** [comment]
@daniyl-x thanks for the hint.

> but resetting podman with podman system reset (back up your volumes, containers, etc.) resolved it for me

Is this _instead of_ adding `network_mode: pasta` per your previous message? For me that single line solved the networking issues in the container.

**Comment by daniyl-x** [comment]
@laur89 yes, this was the solution for me, while `network_mode` is more of a workaround.

If I remember correctly, my issue with `network_mode: pasta` is it only works for single container, and it is not possible to use pasta as a driver (see [podman network create](https://docs.podman.io/en/latest/markdown/podman-network-create.1.html#driver-d-driver)) for your custom network.

If you will use `podman system reset`, you may want to look at `podman export/import`, `podman volume export/import` and `podman image save/load` first, these are basically commands to save container/volume/image into the file and restore from it.
